dropshot/src/handler.rs

```diff
@@ -486,9 +512,18 @@ where
          * resolved statically.
          */
         let rqctx = Arc::new(rqctx_raw);
+        let response_rqctx = rqctx.clone();
         let funcparams = Extractor::from_request(Arc::clone(&rqctx)).await?;
         let future = self.handler.handle_request(rqctx, funcparams);
-        future.await
+
+        future.await.map(|mut response| {
+            let mutable_state = response_rqctx.mutable_state.lock().unwrap();
+
+            for (header, value) in &mutable_state.response_headers {
```

Comment:
I just realized: is this too late?  I'm trying to work this out myself and I'm not sure.

Basically, I'm wondering: is it possible that we've already started streaming the response at this point?  What happens if so?

Reply:
How could we be streaming it out? This function returns it here:
https://github.com/oxidecomputer/dropshot/blob/main/dropshot/src/server.rs#L766
I get that it's called  from an `impl Future`
https://github.com/oxidecomputer/dropshot/blob/main/dropshot/src/server.rs#L766
but we're still going to `await` until we have a an actual `Response<Body>` in hand.

Reply:
I think this is fine.

I'd gotten some wires crossed.  I was thinking that if you've got a whole `Response`, then you've already serialized the body, and in a server that supports streaming, once you've serialized the body, it's too late to add headers.  But having a `Response` doesn't mean you've serialized the response body -- the body can be a Stream whose bytes are read later.